src/lib/TelemetryClient.ts

```diff
@@ -349,7 +349,22 @@ class TelemetryClient {
      * then the specified props will be updated.  Otherwise, a new instance will returned. 
```

Comment:
As long as the expected usage is clear (and not widespread), it's probably fine.  It just struck me as a "JavaScript usually doesn't work this way" kind of a thing and so I had to spend extra time to work out the intent behind it.